app/src/services/sync.js

```diff
@@ -175,7 +175,43 @@ const service = {
 
         // Case: not in COMS - insert new COMS object
         else {
-          const objId = await service._deriveObjectId(s3Object, path, bucketId);
+          let objId = await service._deriveObjectId(s3Object, path, bucketId);
+          let existingComsId;
+          try {
+            existingComsId = await objectService.read(objId);
+          } catch {
+            // Derived object id not currently in use by COMS
+          }
+
+          // Object id in 'coms-id' S3 tag already in use in COMS:
```

Comment:
maybe tidy up the comments?

Reply:
Implementation has since changed - previously this was all in `syncObject()`, it's now mainly handled in `_deriveObjectId()`